### PR TITLE
[MXNET-103] Added tutorial on types of data augmentations.

### PR DESCRIPTION
**Pull request by thomelane**
## Description ##

Added tutorial on types of data augmentations (both augmentations that effect position and color).

### Essentials ###

N/A, as only added a single markdown file.

### Changes ###

Added tutorial.

## Comments ##

Contains images of each type of augmentation.
Contain asserts on array shapes, to allow future testing.
Intend to follow up with another tutorial on how to use Augmenters.

### CODE REVIEW

**Comment by thomelane** [comment]
@cjolivier01 added at https://issues.apache.org/jira/browse/MXNET-103

**Comment by piiswrong** [comment]
We've moved to Gluon augmenters, see https://github.com/apache/incubator-mxnet/pull/10155

**Comment by thomelane** [comment]
@piiswrong I'll create a Gluon version of this when the Gluon augmenters are merged and docs have been added. I still think this is a useful reference for the mx.image Augmenters, and this will be put in the Module section of tutorials.

**Comment by thomelane** [comment]
One of the main aims of this tutorial was to visually show the effects of each augmentation, so I expect this is unchanged from the Gluon transforms?

**Comment by thomelane** [comment]
@rahul003 @szha @piiswrong can we get this merged?

File: docs/tutorials/python/types_of_data_augmentation.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+assert aug_image.shape == (50, 50, 3)
+```
+
+
+![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/data_aug/outputs/types_of/output_55_0.png)
+
+
+# Custom Augmenters
+
+And if you're still hungry for more augmenters, it's easy to perform your own augmentation steps. Just create a function that performs the required steps, and use within the `transform` function of [`mxnet.gluon.data.vision.ImageRecordDataset`](https://mxnet.incubator.apache.org/api/python/gluon/data.html#mxnet.gluon.data.vision.datasets.ImageRecordDataset) or [`ImageFolderDataset`](https://mxnet.incubator.apache.org/api/python/gluon/data.html#mxnet.gluon.data.vision.datasets.ImageFolderDataset). If you're already using Augmenter lists, you can also inherit the `Augmenter` base class, and define your own `__call__` method.
```
Comment (piiswrong) [inline docs/tutorials/python/types_of_data_augmentation.md]: Shouldn't mix gluon in here

**Comment by thomelane** [comment]
@piiswrong I've removed the Gluon `transform` reference and moved the custom augmenters section to the FAQ.

**Pull request merged**